src/auth/jwtaccess.ts

```diff
@@ -57,112 +54,116 @@ export class JWTAccess {
    * Get a non-expired access token, after refreshing if necessary
    *
    * @param {string} authURI the URI being authorized
-   * @param {function} metadataCb a callback invoked with the jwt
-   *                   request metadata.
+   * @param {function} metadataCb a callback invoked with the jwt request metadata.
+   * @returns a Promise that resolves with the request metadata response
    */
-  getRequestMetadata(
-      authURI: string,
-      metadataCb:
-          (err: Error|null, headers?: http.IncomingHttpHeaders|null) => void) {
-    const iat = Math.floor(new Date().getTime() / 1000);
-    const exp = iat + 3600;  // 3600 seconds = 1 hour
+  getRequestMetadata(authURI: string): RequestMetadataResponse;
```

Comment:
Likewise. This method seems synchronous to me. Let's drop the callback version of the API. Open bug (or attach to the running bug for this).

Reply:
added to running issue